### ISSUE TITLE
Issue with nginx and uwsgi

### ISSUE THREAD

**Issue by denisvitez** [rikulo/socket.io-client-dart#329]
Hi,

so I've been struggling to get the Socket.IO connection working with my backend application written in Flask and using Flask-SocketIO for the server.

If I'm hosting my backend server with Python directly, then the connection is working, and I'm connected to the server with:
```
IO.Socket socket = IO.io(
      'http://<MY_IP>:5000',
      IO.OptionBuilder()
          .setPath("/socket.io")
          .setTransports(['websocket'])
          .setExtraHeaders({'Connection': 'upgrade', 'Upgrade': 'websocket'})
          .enableReconnection()
          .setReconnectionDelay(1)
          .enableForceNew()
          .build()
  );
```

If I try to change the IP and connect to the server, I get the "Connection to <SERVER_IP> was not upgraded to websocket" as can be seen with many tickets on here. Since I'm able to connect directly and not on the server I am indeed convinced that the issue is related to Nginx/uWSGI configuration.

So for serving the application on server my configuration is as following:
```
server {
   listen 5000;
   server_name your_domain www.your_domain;

   location / {
       include uwsgi_params;
       uwsgi_pass unix:/nesteh/api/nesteh-api.sock;
   }

   location /socket.io/ {
       proxy_http_version 1.1;
       proxy_set_header Upgrade $http_upgrade;
       proxy_set_header Connection "upgrade";
       include uwsgi_params;
       uwsgi_pass unix:/nesteh/api/nesteh-api.sock;
   }
}
```

This configuration for nginx works, since I'm able to connect to it from my Vue.js application using the `socket.io-client` library without any issues and have been using it for the past 2 years.

However using the Flutter app will result in the "Connection to <SERVER_IP> was not upgraded to websocket" errors.
I've tried changing the nginx config and adding additional parameters as seen on other issues here, but it doesn't seem to make any difference. So now I've end up with the following configuration:
```
server {
    listen 5000;
    server_name _;

    location / {
        include uwsgi_params;
        uwsgi_pass unix:/nesteh/api/nesteh-api.sock;
    }

    location /socket.io/ {
        proxy_pass http://192.168.10.175:5000;
        proxy_redirect off;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        include uwsgi_params;
        uwsgi_pass unix:/nesteh/api/nesteh-api.sock;
    }
}
```

Frontend still works, but nothing changes for the Flutter application.

Does anyone has any ideas what I need to change in my nginx config for it to work? Was anyone able to get this library working with application hosted on uWSGI?

**Comment by denisvitez** [comment]
Ok so I did some digging and I was able to solve it with the help of:
[Flask-SocketIO #1237](https://github.com/miguelgrinberg/Flask-SocketIO/issues/1237)

TL;DR; uWSGI does not support WebSockets, so when hosting an application HTTP mode must be used.
